src/os/file_anyos.go

```diff
@@ -95,7 +96,7 @@ func (fs unixFilesystem) OpenFile(path string, flag int, perm FileMode) (FileHan
 	if flag&O_WRONLY != 0 {
 		syscallFlag |= syscall.O_WRONLY
 	}
-	if flag&O_RDWR != 0 {
+	if flag&O_RDWR == O_RDWR {
```

Comment:
I would be happier if we fixed the rest of the tests to be against the flag value rather than `!= 0`.